source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -82,6 +82,45 @@ enum uma_result_t
 umaMemoryProviderGetLastResult(uma_memory_provider_handle_t hProvider,
                                const char **ppMessage);
 
+///
+/// \brief Retrieve an array of possible pages sizes used by memory region referenced by given ptr
+///        This array is sorted in ascending order.
+/// \param hProvider handle to the memory provider
+/// \param ptr pointer to memory allocated by this memory provider
+/// \param numPageSizes [in] number of memory providers to return. If numPageSizes is not equal to or
+///        greater than the number of possible page sizes, UMA_RESULT_ERROR_INVALID_ARGUMENT is returned.
+/// \param pageSizes [out] will be updated with page size values.
+/// \param numPageSizesRet [in] pointer to the actual number of possible page sizes
+/// \return UMA_RESULT_SUCCESS on success or appropriate error code on failure.
+enum uma_result_t
+umaMemoryProviderGetPageSizes(uma_memory_provider_handle_t hProvider, void *ptr,
```

Comment:
Does the array of possible pageSizes is really needed? Perhaps, just a minimal pageSize is enough because other possible pageSizes should be a multiple of the minimum pageSize.

I am thinking should we also care about proper alignment in the `umaMemoryProviderAlloc` function? And should we at all?

Reply:
Yeah, as I think about it more, probably a function to get a minimum page size should be enough for `purge`.

However, a function for recommended page size could also be useful - it could be used to avoid splitting huge pages and also as a hint for the user, what should he pass as an `alignment` in `umaMemoryProviderAlloc ` to (possibly) get huge-pages.

Reply:
For transparent huge pages: yes - linux will just split those pages to 4KB ones.
For huge TLB (or windows huge pages): no - madvise will return an error.

But for huge TLB an appropriate option needs to be passed during mmap so the memory provider can either track those pages or we can have a memory provider dedicated to huge pages (in that case, it would always return 2MB or 1GB for minPageSize)